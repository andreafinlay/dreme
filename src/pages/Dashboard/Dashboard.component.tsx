import React, { useContext } from 'react';
import { UISref } from '@uirouter/react';
import { RootContext } from '../..';
import { Entries } from '../../components/Entries';
import { Users } from '../../components/Users';

const Dashboard: React.FC<any> = () => {
    let { authBody } = useContext(RootContext);

    return (
        <div>
            <div className='bg-dreme flex'>
                <header>Zzz</header>
                {authBody && <div>{authBody.name}</div>}
                <UISref to='journal'>
                    <a>Journal</a>
                </UISref>
            </div>
            <Entries />
            <Users />
        </div>
    );
};

export { Dashboard };
