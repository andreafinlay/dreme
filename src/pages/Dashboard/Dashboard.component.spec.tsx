import React from 'react';
import { shallow } from 'enzyme';
import { Dashboard } from './Dashboard.component';

describe('init', () => {
    const component = shallow(<Dashboard />);

    it('should render correctly', () => {
        expect(component).toBeDefined();
        expect(component).toMatchSnapshot();
    });

    it('should set initial state', () => {
        expect(component.state()).toEqual(null);
    });
});
