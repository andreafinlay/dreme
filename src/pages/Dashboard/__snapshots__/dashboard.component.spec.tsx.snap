// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`init should render correctly 1`] = `
<div>
  <div
    className="bg-dreme"
  >
    <header>
      Zzz
    </header>
    <UISref
      to="journal"
    >
      <a>
        Journal
      </a>
    </UISref>
  </div>
  <Entries />
</div>
`;
